compiler/vm/vmgen.nim

```diff
@@ -205,8 +205,11 @@ func underlyingLoc(n: CgNode): PSym =
   var root {.cursor.} = n
   # skip nodes that don't change the location until we arrive at either one
   # that does, or a symbol
-  while root.kind in {cnkConv, cnkStmtListExpr}:
-    root = root.lastSon
+  while true:
```

Comment:
```suggestion
  while root.kind in {cnkConv, cnkStmtListExpr}:
```
at first i thought this might be a good idea, but it ends up meaning two places have to be updated correctly in order for things to work. thinking about it a bit further and i realize that i'm suggesting this to avoid leaving infinite loops around.

separate from this PR, do you think we should have an idiom, like so?

```nim
template loopForever(body: untyped) =
  var i = 0
  const maxIterations = 1000
  while i < maxIterations:
    body
    inc i
  doAssert i < maxIterations, "fixed point calculation exceeded maximum iteration attempts"
```

Reply:
Yea, this did bother me a bit too. A `while true` is always a bit meh, though I think that, just like you said, not duplicating the stats is the better choice here.

Regarding the idiom, I'm a bit unsure. To me, it feels like a work-around to missing static analysis -- essentially, the compiler could (and I think should) detect `while` loops where a control-flow path exists on which no progress is made (i.e., no state is modified that is used in conditions guarding loop exits).

Adding the idiom likely still makes sense, but figuring out a `maxIterations` value that works well across different use cases could be hard.

Reply:
Thanks for the input. Static analysis + some wise API design choices is definitely the way to go. In the meantime, we can make the `maxIteration` a parameter (tacked on the end so it doesn't always have to be specified).